### ISSUE TITLE
GUID Primary Key being updated by Database not using v3 POCO Generator Guid.NewGuid()

### ISSUE THREAD

**Issue by c5racing** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#638]
In the v2 generator, I never had a problem with setting the Primary Key of an object when it was being instantiated at the c# level and the database using it.  With v3, even though the ID is set, the database is overwriting it and returning the new Id.  Here is my table:
```sql
CREATE TABLE [dbo].[Test]
(
	[TestId] UNIQUEIDENTIFIER NOT NULL DEFAULT NEWID(),
	[Name] NVARCHAR(60) NOT NULL,
        CONSTRAINT [PkTest] PRIMARY KEY ([TestId]))
)
```

Is there a setting in v3 that change this behavior?

**Comment by c5racing** [comment]
I found the issue.  This was occurring because we utilize our own method in c# for generating Sequential Guids.  This method is called NewSequentialId.  We updated the POCO Generator to use this method vs Guid.NewId().  When the POCOs are being generated, it's looking at the c# default, not the database default.  I'm not sure if this is a bug or to be expected.  Our C# method matched NewSequentialId, so it was adding the Identity.

I changed NewSequentialId to NewSequentialGuid and it works fine now.
